pkg/asset/installconfig/azure/validation.go

```diff
@@ -319,3 +327,20 @@ func validateResourceGroup(client API, fieldPath *field.Path, platform *aztypes.
 	}
 	return allErrs
 }
+
+func validateAzureStackClusterOSImage(StorageEndpointSuffix string, ClusterOSImage string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if ClusterOSImage == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("clusterOSImage"), "clusterOSImage must be set when installing on Azure Stack"))
+	} else {
```

Comment:
This should be the only check that is run in validateForProvisioning.

The other checks should be performed in the main Validate function of install config asset: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/azure/validation.go#L38

That validate function already has a client where you can grab the environment